api/v1alpha3/awscluster_webhook.go

```diff
@@ -29,3 +37,45 @@ func (r *AWSCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awscluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=awsclusters,versions=v1alpha3,name=validation.awscluster.infrastructure.cluster.x-k8s.io
+
+var _ webhook.Validator = &AWSCluster{}
+
+func (r *AWSCluster) ValidateCreate() error {
+	return nil
+}
+
+func (r *AWSCluster) ValidateDelete() error {
+	return nil
+}
+
+func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
+	var allErrs field.ErrorList
+	oldC, ok := old.(*AWSCluster)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected an AWSCluster but got a %T", old))
+	}
+	if r.Spec.Region != oldC.Spec.Region {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "region"), r.Spec.Region, "field is immutable"),
+		)
+	}
+	if !reflect.DeepEqual(r.Spec.ControlPlaneLoadBalancer, oldC.Spec.ControlPlaneLoadBalancer) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "controlPlaneLoadBalancer"), r.Spec.ControlPlaneLoadBalancer, "field is immutable"),
+		)
+	}
+	if !reflect.DeepEqual(oldC.Spec.ControlPlaneEndpoint, clusterv1.APIEndpoint{}) && !reflect.DeepEqual(r.Spec.ControlPlaneEndpoint, oldC.Spec.ControlPlaneEndpoint) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "controlPlaneEndpoint"), r.Spec.ControlPlaneEndpoint, "field is immutable"),
+		)
+	}
+	if oldC.Spec.Bastion.Enabled && !r.Spec.Bastion.Enabled {
```

Comment:
I think we can allow changes to bastion enabled. There is code in `ReconcileBastion` to delete it if enabled is false and we have details about the bastion in awscluster.status.

Reply:
I wonder if we should file an issue to also check by tags to avoid potential orphaning of a bastion if status needs to be recreated for some reason.

Reply:
Are we not already doing that? I assumed we were, since move/pivot can't transfer status.

Reply:
I meant using a tag based lookup rather than relying on `Status.Bastion` to determine if we should delete a previously created bastion host when `Spec.Bastion.Enabled == false`

Reply:
Status.Bastion is populated by a tag lookup, and the deletion appears to also use the same tag lookup to determine the instance to terminate. So I think we're good?

Reply:
Wouldn't that require that `AWSCluster` be reconciled once w/ `Spec.Bastion.Enabled = true` to update the status prior to being able to delete the bastion in the case that status is cleared for some reason? It seems like if for some reason between reconciliation attempts both the Status is cleared and `Spec.Bastion.Enabled` changes from `false` to `true` that we would still orphan the instance (and also block the eventual successful deletion of the AWSCluster).

Reply:
Yeah, you're right. Could we adjust the logic in ReconcileBastion to describe first, then process a deletion if necessary?

Reply:
I created #1678 to track the issue.

Reply:
@benmoss based on the conversation above, lets go ahead and remove the validation preventing disabling of an existing bastion.